// @opentelemetry/instrumentation-openai/build/src/responses.js
"use strict";
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConvertResponseOutputsToOutputMessagesUseCase = exports.ConvertResponseInputsToInputMessagesUseCase = void 0;
class ConvertResponseInputsToInputMessagesUseCase {
    captureMessageContent;
    constructor(captureMessageContent = false) {
        this.captureMessageContent = captureMessageContent;
    }
    convert(params) {
        const messages = [];
        if (typeof params.instructions === 'string') {
            messages.push(this.message({ role: 'system', content: params.instructions }));
        }
        if (typeof params.input === 'string') {
            messages.push(this.message({ role: 'user', content: params.input }));
        }
        else if (Array.isArray(params.input)) {
            messages.push(...params.input.map((input) => this[input.type ?? 'message'](input)));
        }
        return messages;
    }
    message(item) {
        const parts = [];
        if (typeof item.content === 'string') {
            if (this.captureMessageContent) {
                parts.push({
                    type: 'text',
                    content: item.content,
                });
            }
            else {
                parts.push({
                    type: 'text',
                    content: undefined,
                });
            }
        }
        else if (Array.isArray(item.content)) {
            for (const content of item.content) {
                switch (content.type) {
                    case 'input_text':
                    case 'output_text':
                        if (this.captureMessageContent) {
                            parts.push({
                                type: 'text',
                                content: content.text,
                            });
                        }
                        else {
                            parts.push({
                                type: 'text',
                                content: undefined,
                            });
                        }
                        break;
                    case 'refusal':
                        parts.push({
                            type: 'refusal',
                            refusal: content.refusal,
                        });
                        break;
                    case 'input_image':
                        parts.push({
                            ...(this.captureMessageContent ? content : undefined),
                            type: 'image',
                        });
                        break;
                    case 'input_file':
                        parts.push({
                            ...(this.captureMessageContent ? content : undefined),
                            type: 'file',
                        });
                        break;
                    default: {
                        parts.push({
                            ...(this.captureMessageContent ? content : undefined),
                            type: 'audio',
                        });
                        break;
                    }
                }
            }
        }
        return {
            role: item.role,
            parts,
        };
    }
    function_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.name,
                    arguments: this.captureMessageContent ? item.arguments : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    custom_tool_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.name,
                    arguments: this.captureMessageContent ? item.input : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    reasoning(item) {
        const parts = [];
        for (const summary of item.summary) {
            parts.push({
                type: item.type,
                text: this.captureMessageContent ? summary.text : undefined,
            });
        }
        if (item.content) {
            for (const content of item.content) {
                parts.push({
                    type: item.type,
                    text: this.captureMessageContent ? content.text : undefined,
                });
            }
        }
        return {
            role: 'assistant',
            parts,
        };
    }
    file_search_call(item) {
        const parts = [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.queries : undefined,
            },
        ];
        for (const result of item.results ?? []) {
            parts.push({
                type: 'tool_call_response',
                id: item.id,
                response: this.captureMessageContent ? result : undefined,
            });
        }
        return {
            role: 'assistant',
            parts,
        };
    }
    web_search_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.type,
                    arguments: this.captureMessageContent ? item.action : undefined,
                },
            ],
        };
    }
    computer_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.type,
                    arguments: this.captureMessageContent ? item.action : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    computer_call_output(item) {
        return {
            role: 'user',
            parts: [
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: this.captureMessageContent ? item.output : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    code_interpreter_call(item) {
        const parts = [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.code : undefined,
            },
        ];
        for (const output of item.outputs ?? []) {
            switch (output.type) {
                case 'image':
                    parts.push({
                        type: 'tool_call_response',
                        id: item.id,
                        response: this.captureMessageContent ? output.url : undefined,
                    });
                    break;
                case 'logs':
                    parts.push({
                        type: 'tool_call_response',
                        id: item.id,
                        response: this.captureMessageContent ? output.logs : undefined,
                    });
                    break;
            }
        }
        return {
            role: 'assistant',
            parts,
        };
    }
    image_generation_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.type,
                },
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: this.captureMessageContent ? item.result : undefined,
                },
            ],
        };
    }
    function_call_output(item) {
        return {
            role: 'user',
            parts: [
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: this.captureMessageContent ? item.output : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    local_shell_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.type,
                    arguments: this.captureMessageContent ? item.action : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    local_shell_call_output(item) {
        return {
            role: 'user',
            parts: [
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: this.captureMessageContent ? item.output : undefined,
                },
            ],
        };
    }
    mcp_call(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: item.type,
                    arguments: this.captureMessageContent
                        ? `${item.name}(${item.arguments})`
                        : undefined,
                    server: item.server_label,
                },
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: item.error
                        ? item.error
                        : this.captureMessageContent
                            ? item.output
                            : undefined,
                    server: item.server_label,
                },
            ],
        };
    }
    mcp_list_tools(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: item.error
                        ? item.error
                        : this.captureMessageContent
                            ? item.tools
                            : undefined,
                    server: item.server_label,
                },
            ],
        };
    }
    mcp_approval_request(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'tool_call',
                    id: item.id,
                    name: `${item.type}${this.captureMessageContent ? `: ${item.name}` : ''}`,
                    arguments: this.captureMessageContent ? item.arguments : undefined,
                    server: item.server_label,
                },
            ],
        };
    }
    mcp_approval_response(item) {
        return {
            role: 'user',
            parts: [
                {
                    type: 'tool_call_response',
                    response: this.captureMessageContent ? item.approve : undefined,
                },
            ],
        };
    }
    custom_tool_call_output(item) {
        return {
            role: 'user',
            parts: [
                {
                    type: 'tool_call_response',
                    id: item.id,
                    response: this.captureMessageContent ? item.output : undefined,
                    call_id: item.call_id,
                },
            ],
        };
    }
    item_reference(item) {
        return {
            role: 'assistant',
            parts: [
                {
                    type: 'item_reference',
                    id: item.id,
                },
            ],
        };
    }
}
exports.ConvertResponseInputsToInputMessagesUseCase = ConvertResponseInputsToInputMessagesUseCase;
class ConvertResponseOutputsToOutputMessagesUseCase {
    captureMessageContent;
    constructor(captureMessageContent = false) {
        this.captureMessageContent = captureMessageContent;
    }
    convert(responseOutput) {
        const parts = responseOutput.flatMap((item) => this[item.type](item));
        return [
            {
                role: 'assistant',
                parts,
                finish_reason: parts[parts.length - 1]?.type === 'tool_call' ? 'tool_call' : 'stop',
            },
        ];
    }
    message(item) {
        const parts = [];
        for (const content of item.content) {
            switch (content.type) {
                case 'output_text':
                    if (this.captureMessageContent) {
                        parts.push({
                            type: 'text',
                            content: content.text,
                        });
                    }
                    else {
                        parts.push({
                            type: 'text',
                            content: undefined,
                        });
                    }
                    break;
                case 'refusal':
                    parts.push({
                        type: content.type,
                        refusal: content.refusal,
                    });
                    break;
            }
        }
        return parts;
    }
    function_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.name,
                arguments: this.captureMessageContent ? item.arguments : undefined,
                call_id: item.call_id,
            },
        ];
    }
    custom_tool_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.name,
                arguments: this.captureMessageContent ? item.input : undefined,
                call_id: item.call_id,
            },
        ];
    }
    reasoning(item) {
        const parts = [];
        for (const summary of item.summary) {
            parts.push({
                type: item.type,
                text: this.captureMessageContent ? summary.text : undefined,
            });
        }
        if (item.content) {
            for (const content of item.content) {
                parts.push({
                    type: item.type,
                    text: this.captureMessageContent ? content.text : undefined,
                });
            }
        }
        return parts;
    }
    file_search_call(item) {
        const parts = [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.queries : undefined,
            },
        ];
        for (const result of item.results ?? []) {
            parts.push({
                type: 'tool_call_response',
                id: item.id,
                response: this.captureMessageContent ? result : undefined,
            });
        }
        return parts;
    }
    web_search_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.action : undefined,
            },
        ];
    }
    computer_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.action : undefined,
                call_id: item.call_id,
            },
        ];
    }
    code_interpreter_call(item) {
        const parts = [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.code : undefined,
            },
        ];
        for (const output of item.outputs ?? []) {
            switch (output.type) {
                case 'image':
                    parts.push({
                        type: 'tool_call_response',
                        id: item.id,
                        response: this.captureMessageContent ? output.url : undefined,
                    });
                    break;
                case 'logs':
                    parts.push({
                        type: 'tool_call_response',
                        id: item.id,
                        response: this.captureMessageContent ? output.logs : undefined,
                    });
                    break;
            }
        }
        return parts;
    }
    image_generation_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
            },
            {
                type: 'tool_call_response',
                id: item.id,
                response: this.captureMessageContent ? item.result : undefined,
            },
        ];
    }
    local_shell_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.type,
                arguments: this.captureMessageContent ? item.action : undefined,
                call_id: item.call_id,
            },
        ];
    }
    mcp_call(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: item.name,
                arguments: this.captureMessageContent
                    ? `${item.name}(${item.arguments})`
                    : undefined,
                server: item.server_label,
            },
            {
                type: 'tool_call_response',
                id: item.id,
                response: item.error
                    ? item.error
                    : this.captureMessageContent
                        ? item.output
                        : undefined,
                server: item.server_label,
            },
        ];
    }
    mcp_list_tools(item) {
        return [
            {
                type: 'tool_call_response',
                id: item.id,
                response: item.error
                    ? item.error
                    : this.captureMessageContent
                        ? item.tools
                        : undefined,
                server: item.server_label,
            },
        ];
    }
    mcp_approval_request(item) {
        return [
            {
                type: 'tool_call',
                id: item.id,
                name: `${item.type}${this.captureMessageContent ? `: ${item.name}` : ''}`,
                arguments: this.captureMessageContent ? item.arguments : undefined,
                server: item.server_label,
            },
        ];
    }
}
exports.ConvertResponseOutputsToOutputMessagesUseCase = ConvertResponseOutputsToOutputMessagesUseCase;
