// @opentelemetry/configuration/build/src/FileConfigFactory.js
"use strict";
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.mergePropagatorCompositeConfig = exports.mergeResourceAttributesConfig = exports.parseConfigFile = exports.FileConfigFactory = void 0;
const api_1 = require("@opentelemetry/api");
const core_1 = require("@opentelemetry/core");
const fs = require("fs");
const yaml = require("yaml");
const utils_1 = require("./utils");
// Pre-compiled ajv validator — eliminates runtime schema compilation on cold start.
// Generated by `npm run generate:config`; do not import ajv directly here.
const validateConfig = require("./generated/validator");
class FileConfigFactory {
    _config;
    constructor() {
        this._config = parseConfigFile();
    }
    getConfigModel() {
        return this._config;
    }
}
exports.FileConfigFactory = FileConfigFactory;
// The schema version this SDK targets. Per the configuration versioning spec,
// an implementation MUST reject a config file whose MAJOR version differs (a
// breaking-change boundary), accepts any MINOR version within that major, and
// SHOULD warn when the MINOR version is newer than it targets since unknown
// settings may be ignored. See
// https://github.com/open-telemetry/opentelemetry-configuration/blob/main/VERSIONING.md
const SUPPORTED_FILE_FORMAT_MAJOR = 1;
const SUPPORTED_FILE_FORMAT_MINOR = 1;
function validateFileFormat(configFile, fileFormat) {
    if (typeof fileFormat !== 'string' || fileFormat.length === 0) {
        throw new Error(`${configFile}: Unsupported file_format: "${fileFormat}". Expected a "MAJOR.MINOR" version string.`);
    }
    // Drop any pre-release / meta tag, e.g. "1.0-rc.2" -> "1.0".
    const parts = fileFormat.split('-', 1)[0].split('.');
    const major = Number(parts[0]);
    const minor = parts.length > 1 ? Number(parts[1]) : 0;
    if (!Number.isInteger(major) || !Number.isInteger(minor)) {
        throw new Error(`${configFile}: Unsupported file_format: "${fileFormat}". Expected "MAJOR.MINOR" version numbers.`);
    }
    if (major !== SUPPORTED_FILE_FORMAT_MAJOR) {
        throw new Error(`${configFile}: Unsupported file_format: "${fileFormat}". This SDK supports schema version ${SUPPORTED_FILE_FORMAT_MAJOR}.x.`);
    }
    if (minor > SUPPORTED_FILE_FORMAT_MINOR) {
        api_1.diag.warn(`${configFile}: Configuration file_format "${fileFormat}" has a newer minor version than this SDK supports (${SUPPORTED_FILE_FORMAT_MAJOR}.${SUPPORTED_FILE_FORMAT_MINOR}); some settings may be ignored.`);
    }
}
function parseConfigFile() {
    const configFile = (0, core_1.getStringFromEnv)('OTEL_CONFIG_FILE') || '';
    const file = fs.readFileSync(configFile, 'utf8');
    const doc = yaml.parseDocument(file, { version: '1.2' });
    (0, utils_1.substituteEnvVars)(doc);
    const processed = doc.toJS();
    validateFileFormat(configFile, processed?.file_format);
    const valid = validateConfig(processed);
    if (!valid) {
        let detail;
        if (!validateConfig.errors) {
            detail = 'unknown error';
        }
        else if (validateConfig.errors.length === 1) {
            const err = validateConfig.errors[0];
            detail = `${err.instancePath} ${err.message}`;
        }
        else {
            const sep = '\n  ';
            detail =
                sep +
                    validateConfig.errors
                        .map(e => `${e.instancePath} ${e.message}`)
                        .join(sep);
        }
        throw new Error(`Invalid OpenTelemetry config file: ${configFile}: ${detail}`);
    }
    const data = processed;
    // Strip file_format from output — it's a meta-field, not a config value
    delete data['file_format'];
    return data;
}
exports.parseConfigFile = parseConfigFile;
/**
 * Merge attributes from `resource.attributes` and `resource.attributes_list`
 * (comma-separated key=value pairs) into an `AttributeNameValue[]` array.
 * Entries already in `resource.attributes` taking precedence.
 *
 * Note: The returned array might not be a copy, so it should not be mutated.
 */
function mergeResourceAttributesConfig(attributes, attributes_list) {
    if (!attributes_list) {
        return attributes;
    }
    const mergedAttrs = attributes ? attributes.slice() : [];
    const existingNames = new Set(mergedAttrs.map(a => a.name));
    // TODO(6842): handle attribute limits, if any
    let discard = false;
    const listAttrs = {};
    for (const pair of attributes_list.split(',')) {
        if (pair.trim() === '')
            continue;
        const parts = pair.split('=');
        if (parts.length !== 2) {
            api_1.diag.warn(`Invalid format for resource.attributes_list entry "${pair}": expected key=value with '=' percent-encoded in keys/values. Discarding all entries.`);
            discard = true;
            break;
        }
        const rawName = parts[0].trim();
        const rawValue = parts[1].trim();
        if (rawName === '') {
            api_1.diag.warn(`Empty attribute name in resource.attributes_list entry "${pair}". Discarding all entries.`);
            discard = true;
            break;
        }
        try {
            listAttrs[decodeURIComponent(rawName)] = decodeURIComponent(rawValue);
        }
        catch (e) {
            api_1.diag.warn(`Failed to percent-decode resource.attributes_list entry "${pair}", discarding all entries: ${e}`);
            discard = true;
            break;
        }
    }
    if (!discard) {
        for (const [name, value] of Object.entries(listAttrs)) {
            if (!existingNames.has(name)) {
                mergedAttrs.push({ name, value, type: 'string' });
            }
        }
    }
    return mergedAttrs;
}
exports.mergeResourceAttributesConfig = mergeResourceAttributesConfig;
/**
 * Merge propagator.composite_list (comma-separated propagator names) into
 * propagator.composite, with entries already in composite taking precedence.
 * Merge TextMapPropagator configs from `propagator.composite` and `propagator.composite_list`
 * (comma-separated names) into a `TextMapPropagator[]` array.
 * Entries already in `propagator.composite` taking precedence.
 *
 * Note: The returned array might not be a copy, so it should not be mutated.
 */
function mergePropagatorCompositeConfig(composite, composite_list) {
    if (!composite_list) {
        return composite;
    }
    const mergedComposite = composite ? composite.slice() : [];
    const existingNames = new Set(mergedComposite.map(c => Object.keys(c)[0]));
    for (const name of composite_list.split(',')) {
        const trimmed = name.trim();
        if (trimmed && !existingNames.has(trimmed)) {
            mergedComposite.push({ [trimmed]: null });
            existingNames.add(trimmed);
        }
    }
    return mergedComposite;
}
exports.mergePropagatorCompositeConfig = mergePropagatorCompositeConfig;
